src/plugins/gui/creditCardPlugin.js

```diff
@@ -161,11 +166,47 @@ export const creditCardPlugin: FiatPluginFactory = async (params: FiatPluginFact
             return
           }
 
-          const bestRate = div(bestQuote.fiatAmount, bestQuote.cryptoAmount, 16)
-          const exchangeRateText = `1 ${currencyCode} = ${toFixed(bestRate, 0, 2)} ${FIAT_SUPPORTED}`
+          const exchangeRateText = getRateFromQuote(bestQuote)
           if (enterAmountMethods != null) {
+            const poweredByOnClick = async () => {
+              // 1. Show modal with all the valid quotes
+              const items = goodQuotes.map(quote => {
+                let text
+                if (sourceFieldNum === 1) {
+                  // User entered a fiat value. Show the crypto value per partner
+                  text = `(${toFixed(quote.cryptoAmount, 0, 6)} ${quote.tokenId?.tokenId ?? ''})`
+                } else {
+                  // User entered a crypto value. Show the fiat value per partner
+                  text = `(${toFixed(quote.fiatAmount, 0, 2)} ${quote.fiatCurrencyCode.replace('iso:', '')})`
+                }
+                const out = {
+                  text,
+                  name: quote.pluginDisplayName,
+                  icon: getPartnerIconUri(quote.partnerIcon)
+                }
+                return out
+              })
+              const rowName = await showUi.listModal({
+                title: 'Providers',
+                selected: bestQuote?.pluginDisplayName ?? '',
+                items
+              })
+              if (bestQuote == null) return
+
+              // 2. Set the best quote to the one chosen by user (if any is chosen)
+              if (rowName != null && rowName !== bestQuote.pluginDisplayName) {
+                bestQuote = goodQuotes.find(quote => quote.pluginDisplayName === rowName)
+                if (bestQuote == null) return
+
+                // 3. Set the status text and powered by
+                const statusText = getRateFromQuote(bestQuote)
+                enterAmountMethods.setStatusText({ statusText })
+                enterAmountMethods.setPoweredBy({ poweredByText: bestQuote.pluginDisplayName, poweredByIcon: bestQuote.partnerIcon, poweredByOnClick })
```

Comment:
Oh, I get it - we want this to be interactive, so the user can change the powered-by text by tapping.

I wonder if we can clean this up to avoid the whole `getMethods` round-trip. Maybe the top-level `enterAmount` method returns an `updateProps` method, in addition to the results promise, so we have a single location to change anything we need, including future stuff we might add or existing things like labels.